plugins/inputs/cisco_dna_center/cisco_dna_center.go

```diff
@@ -0,0 +1,176 @@
+package cisco_dna_center
+
+import (
+	"net/url"
+
+	dnac_sdk "github.com/cisco-en-programmability/dnacenter-go-sdk/v3/sdk"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Dnac struct {
+	Username    string
+	Password    string
+	SSLVerify   string
+	Report      []string
+	Log         telegraf.Logger
+	Client      *dnac_sdk.Client
+	DnacBaseURL string
+}
+
+func (d *Dnac) Description() string {
+	return "a Cisco DNAC plugin"
+}
+
+func (d *Dnac) SampleConfig() string {
+	return `
+  ## Specify DNAC Base URL
+  dnacbaseurl = "https://sandboxdnac.cisco.com"
+  ## Specify Credentials
+  username = "devnetuser"
+  password = "Cisco123!"
+  ## SSL Verify true/false
+  sslverify = "false"
+  ## Health types to report
+  report = ["client","network"]
+`
+}
+
+// Init is for setup, and validating config.
+func (d *Dnac) Init() error {
+	err := d.InitClient()
+	if err != nil {
+		return err
+	}
+	err = d.Client.AuthClient()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Dnac) InitClient() error {
+	var debug string
+	if telegraf.Debug {
+		debug = "true"
+	} else {
+		debug = "false"
+	}
+
+	var client, err = dnac_sdk.NewClientWithOptionsNoAuth(d.DnacBaseURL, d.Username, d.Password, debug, d.SSLVerify)
+	if err != nil {
+		d.Log.Errorf("Connection or login to DNAC failed")
+		return err
+	}
+	d.Client = client
+	return nil
+}
+
+func NewDnac() *Dnac {
+	return &Dnac{}
+}
+
+func (d *Dnac) Gather(acc telegraf.Accumulator) error {
+	var err error
+
+	var dnacURL, urlErr = url.Parse(d.DnacBaseURL)
+	if urlErr != nil {
+		d.Log.Errorf("Invalid DNAC Base URL Provided")
+		return urlErr
+	}
+
+	if d.Client == nil {
+		err = d.InitClient()
+		if err != nil {
+			return err
+		}
+	}
+
+	err = d.Client.AuthClient()
+	if err != nil {
+		return err
+	}
+
+	for healthType := range d.Report {
+		if d.Report[healthType] == "client" {
```

Comment:
Use the power of golang! And this should _really_ be a `switch` statement...
```suggestion
	for _, healthType := range d.Report {
		switch healthType {
		case "client":
```